### PR TITLE
fix bug: throw error when md contain wrong path.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @qinezh @ansyral @hellosnow @vicancy

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -211,25 +211,17 @@ private MarkupResult ParseCore(MarkupResult markupResult, FileAndType ft)
...
             {
-                string linkFile;
                 string anchor = null;
                 var link = pair.Attr;
-                if (TypeForwardedToPathUtility.IsRelativePath(link.Value))
+                string linkFile = link.Value;
+                var index = linkFile.IndexOfAny(UriFragmentOrQueryString);
+                if (index != -1)
+                {
+                    anchor = linkFile.Substring(index);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >anchor [](start = 20, length = 6)

why deal with absolute path?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/Path/RelativePath.cs [hunk 2]
```diff
@@ -41,6 +41,15 @@ private RelativePath(bool isFromWorkingFolder, int parentDirectoryCount, string[
 
         #region Public Members
 
+        public static bool IsRelativePath(string path)
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/Path/RelativePath.cs]: >IsRelativePath [](start = 27, length = 14)

add a TODO to merge with the PathUtility one

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: avoid link with query string or fragment.

**Pull request merged**